prow/plugins/projectmanager/projectmanager.go

```diff
@@ -108,61 +115,102 @@ type githubClient interface {
 	CreateProjectCard(columnID int, projectCard github.ProjectCard) (*github.ProjectCard, error)
 }
 
-func handlePR(gc githubClient, projectManager plugins.ProjectManager, log *logrus.Entry, pe github.PullRequestEvent) error {
-	// Only respond to label add or issue/PR open events
-	if pe.Action != github.PullRequestActionOpened &&
-		pe.Action != github.PullRequestActionReopened &&
-		pe.Action != github.PullRequestActionLabeled {
+type eventData struct {
+	id     int
+	number int
+	isPR   bool
+	org    string
+	repo   string
+	state  string
+	labels []github.Label
+	remove bool
+}
+
+type DuplicateCard struct {
+	projectName string
+	issueURL    string
+}
+
+func (m *DuplicateCard) Error() string {
+	return fmt.Sprintf(issueAlreadyInProject, m.issueURL, m.projectName)
+}
+
+func handleIssueRequest(pc plugins.Agent, ie github.IssueEvent) error {
+	if !handleIssueActions[ie.Action] {
+		return nil
+	}
+	eventData := eventData{ie.Issue.ID, ie.Issue.Number, false, ie.Repo.Owner.Login, ie.Repo.Name, ie.Issue.State, ie.Issue.Labels, false}
+	if ie.Action == github.IssueActionUnlabeled {
+		eventData.remove = true
+	}
+
+	return handle(pc.GitHubClient, pc.PluginConfig.ProjectManager, pc.Logger, eventData)
+}
+
+func handlePullRequest(pc plugins.Agent, pe github.PullRequestEvent) error {
+	if !handlePRActions[pe.Action] {
 		return nil
 	}
+	eventData := eventData{pe.PullRequest.ID, pe.Number, true, pe.Repo.Owner.Login, pe.Repo.Name, pe.PullRequest.State, []github.Label{}, false}
```

Comment:
Please specify field names in literal struct declarations so that field ordering isn't significant.

Reply:
Fixed.